src/utils/index.ts

```diff
@@ -0,0 +1,62 @@
+import { Currency, Token } from '@uniswap/sdk-core'
+import { Pair } from '@uniswap/v2-sdk'
+import { Pool } from '@uniswap/v3-sdk'
+import { MixedRouteSDK } from '../entities/mixedRoute/route'
+
+/**
+ * Utility function to return each consecutive section of Pools or Pairs in a MixedRoute
+ * @param route
+ * @returns a nested array of Pools or Pairs in the order of the route
+ */
+export const divideMixedRouteIntoConsecutiveSections = (
```

Comment:
I think this function could be a little cleaner by implementing your own "while" loop by just keeping track of where in the list you are with pointers. I wrote out a version of this that I can send but will first leave it to u as a fun exercise. not high priority but its somewhat hard to read with multiple while loops and an outside if condition that checks the same condition that the inside while loop checks.

Reply:
I agree, pushing up a rewrite in next commit. Feels like a leetcode question lol